<a href="https://colab.research.google.com/github/Pawan-kumar-Singh/Global-Terrorism/blob/main/Global_Terrorism_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Global Terrorism Database (GTD) is an open-source database including information on terrorist attacks around the world from 1970 through 2017. The GTD includes systematic data on domestic as well as international terrorist incidents that have occurred during this time period and now includes more than 180,000 attacks. The database is maintained by researchers at the National Consortium for the Study of Terrorism and Responses to Terrorism (START), headquartered at the University of Maryland.</b>

# <b> Explore and analyze the data to discover key findings pertaining to terrorist activities. </b>

## **Importing Important libraries for analysis**

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
df = pd.read_csv('/content/drive/MyDrive/Global Terrorism Analysis/Global Terrorism Data.csv',encoding="latin1")

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4,6,31,33,61,62,63,76,79,90,92,94,96,114,115,121) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [7]:
df.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... addnotes scite1 scite2  scite3  dbsource  \
0  Dominican Republic       2  ...      NaN    NaN    NaN     NaN      PGIS   
1              Mexico       1  ...      NaN    NaN    NaN     NaN      PGIS   
2         Philippines       5  ...      NaN    NaN    NaN     NaN      PGIS   
3              Greece       8  ...      NaN    NaN    NaN     NaN      PGIS   
4               Japan       4  ...      NaN    NaN    NaN     NaN      PGIS   

   INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0        0         0        0       0      NaN  
1        0         1        1       1      NaN  
2       -9        -9        1       1      NaN  
3       -9        -9        1       1      NaN  
4       -9        -9        1       1      NaN  

[5 rows x 135 columns]

In [8]:
df.shape

(181691, 135)

In [9]:
df.columns

Index(['eventid', 'iyear', 'imonth', 'iday', 'approxdate', 'extended',
       'resolution', 'country', 'country_txt', 'region',
       ...
       'addnotes', 'scite1', 'scite2', 'scite3', 'dbsource', 'INT_LOG',
       'INT_IDEO', 'INT_MISC', 'INT_ANY', 'related'],
      dtype='object', length=135)

In [11]:
for i in df.columns:
  print(i,end=", ")

eventid, iyear, imonth, iday, approxdate, extended, resolution, country, country_txt, region, region_txt, provstate, city, latitude, longitude, specificity, vicinity, location, summary, crit1, crit2, crit3, doubtterr, alternative, alternative_txt, multiple, success, suicide, attacktype1, attacktype1_txt, attacktype2, attacktype2_txt, attacktype3, attacktype3_txt, targtype1, targtype1_txt, targsubtype1, targsubtype1_txt, corp1, target1, natlty1, natlty1_txt, targtype2, targtype2_txt, targsubtype2, targsubtype2_txt, corp2, target2, natlty2, natlty2_txt, targtype3, targtype3_txt, targsubtype3, targsubtype3_txt, corp3, target3, natlty3, natlty3_txt, gname, gsubname, gname2, gsubname2, gname3, gsubname3, motive, guncertain1, guncertain2, guncertain3, individual, nperps, nperpcap, claimed, claimmode, claimmode_txt, claim2, claimmode2, claimmode2_txt, claim3, claimmode3, claimmode3_txt, compclaim, weaptype1, weaptype1_txt, weapsubtype1, weapsubtype1_txt, weaptype2, weaptype2_txt, weapsubtype2

In [13]:
df=df[["iyear","imonth","iday","country_txt","region_txt","provstate","city","latitude","longitude","location","summary","attacktype1_txt","targtype1_txt","gname","motive","weaptype1_txt","nkill","nwound","addnotes"]]

In [14]:
df.head()

iyear  imonth  iday         country_txt                   region_txt  \
0   1970       7     2  Dominican Republic  Central America & Caribbean   
1   1970       0     0              Mexico                North America   
2   1970       1     0         Philippines               Southeast Asia   
3   1970       1     0              Greece               Western Europe   
4   1970       1     0               Japan                    East Asia   

  provstate           city   latitude   longitude location summary  \
0       NaN  Santo Domingo  18.456792  -69.951164      NaN     NaN   
1   Federal    Mexico city  19.371887  -99.086624      NaN     NaN   
2    Tarlac        Unknown  15.478598  120.599741      NaN     NaN   
3    Attica         Athens  37.997490   23.762728      NaN     NaN   
4   Fukouka        Fukouka  33.580412  130.396361      NaN     NaN   

                  attacktype1_txt                targtype1_txt  \
0                   Assassination  Private Citizens & Property   
1     Hostage Taking (Kidnapping)      Government (Diplomatic)   
2                   Assassination          Journalists & Media   
3               Bombing/Explosion      Government (Diplomatic)   
4  Facility/Infrastructure Attack      Government (Diplomatic)   

                                gname motive weaptype1_txt  nkill  nwound  \
0                              MANO-D    NaN       Unknown    1.0     0.0   
1  23rd of September Communist League    NaN       Unknown    0.0     0.0   
2                             Unknown    NaN       Unknown    1.0     0.0   
3                             Unknown    NaN    Explosives    NaN     NaN   
4                             Unknown    NaN    Incendiary    NaN     NaN   

  addnotes  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN

In [16]:
df.rename(columns={"iyear":"Year","imonth":"Month","iday":"Day","country_txt":"Country","region_txt":"Region","provstate":"ProvinceOrState","city":"City","latitude":"Latitude","longitude":"Longitude","location":"Location","summary":"Summary","attacktype1_txt":"Attack_Type","targtype1_txt":"Target_Type","gname":"Group_Name","motive":"Motive","weaptype1_txt":"Weapon_Type","nkill":"No_Of_Killed","nwound":"No_Of_Wounded","addnotes":"Add_Notes"},inplace=True)

/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py:5047: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  errors=errors,


In [17]:
df.head()

Year  Month  Day             Country                       Region  \
0  1970      7    2  Dominican Republic  Central America & Caribbean   
1  1970      0    0              Mexico                North America   
2  1970      1    0         Philippines               Southeast Asia   
3  1970      1    0              Greece               Western Europe   
4  1970      1    0               Japan                    East Asia   

  ProvinceOrState           City   Latitude   Longitude Location Summary  \
0             NaN  Santo Domingo  18.456792  -69.951164      NaN     NaN   
1         Federal    Mexico city  19.371887  -99.086624      NaN     NaN   
2          Tarlac        Unknown  15.478598  120.599741      NaN     NaN   
3          Attica         Athens  37.997490   23.762728      NaN     NaN   
4         Fukouka        Fukouka  33.580412  130.396361      NaN     NaN   

                      Attack_Type                  Target_Type  \
0                   Assassination  Private Citizens & Property   
1     Hostage Taking (Kidnapping)      Government (Diplomatic)   
2                   Assassination          Journalists & Media   
3               Bombing/Explosion      Government (Diplomatic)   
4  Facility/Infrastructure Attack      Government (Diplomatic)   

                           Group_Name Motive Weapon_Type  No_Of_Killed  \
0                              MANO-D    NaN     Unknown           1.0   
1  23rd of September Communist League    NaN     Unknown           0.0   
2                             Unknown    NaN     Unknown           1.0   
3                             Unknown    NaN  Explosives           NaN   
4                             Unknown    NaN  Incendiary           NaN   

   No_Of_Wounded Add_Notes  
0            0.0       NaN  
1            0.0       NaN  
2            0.0       NaN  
3            NaN       NaN  
4            NaN       NaN

In [18]:
df.tail()

Year  Month  Day      Country                      Region  \
181686  2017     12   31      Somalia          Sub-Saharan Africa   
181687  2017     12   31        Syria  Middle East & North Africa   
181688  2017     12   31  Philippines              Southeast Asia   
181689  2017     12   31        India                  South Asia   
181690  2017     12   31  Philippines              Southeast Asia   

        ProvinceOrState           City   Latitude   Longitude  \
181686  Middle Shebelle  Ceelka Geelow   2.359673   45.385034   
181687         Lattakia         Jableh  35.407278   35.942679   
181688      Maguindanao       Kubentog   6.900742  124.437908   
181689          Manipur         Imphal  24.798346   93.940430   
181690      Maguindanao  Cotabato City   7.209594  124.241966   

                                                 Location  \
181686     The incident occurred near the town of Balcad.   
181687     The incident occurred at the Humaymim Airport.   
181688  The incident occurred in the Datu Hoffer distr...   
181689  The incident occurred in the Mantripukhri neig...   
181690                                                NaN   

                                                  Summary  \
181686  12/31/2017: Assailants opened fire on a Somali...   
181687  12/31/2017: Assailants launched mortars at the...   
181688  12/31/2017: Assailants set fire to houses in K...   
181689  12/31/2017: Assailants threw a grenade at a Fo...   
181690  12/31/2017: An explosive device was discovered...   

                           Attack_Type                  Target_Type  \
181686                   Armed Assault                     Military   
181687               Bombing/Explosion                     Military   
181688  Facility/Infrastructure Attack  Private Citizens & Property   
181689               Bombing/Explosion         Government (General)   
181690               Bombing/Explosion                      Unknown   

                                        Group_Name Motive Weapon_Type  \
181686                                  Al-Shabaab    NaN    Firearms   
181687                           Muslim extremists    NaN  Explosives   
181688  Bangsamoro Islamic Freedom Movement (BIFM)    NaN  Incendiary   
181689                                     Unknown    NaN  Explosives   
181690                                     Unknown    NaN  Explosives   

        No_Of_Killed  No_Of_Wounded Add_Notes  
181686           1.0            2.0       NaN  
181687           2.0            7.0       NaN  
181688           0.0            0.0       NaN  
181689           0.0            0.0       NaN  
181690           0.0            0.0       NaN

In [19]:
df.shape

(181691, 19)

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Data columns (total 19 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Year             181691 non-null  int64  
 1   Month            181691 non-null  int64  
 2   Day              181691 non-null  int64  
 3   Country          181691 non-null  object 
 4   Region           181691 non-null  object 
 5   ProvinceOrState  181270 non-null  object 
 6   City             181257 non-null  object 
 7   Latitude         177135 non-null  float64
 8   Longitude        177134 non-null  float64
 9   Location         55495 non-null   object 
 10  Summary          115562 non-null  object 
 11  Attack_Type      181691 non-null  object 
 12  Target_Type      181691 non-null  object 
 13  Group_Name       181691 non-null  object 
 14  Motive           50561 non-null   object 
 15  Weapon_Type      181691 non-null  object 
 16  No_Of_Killed     171378 non-null  floa

In [23]:
df.isnull().sum()

Year                    0
Month                   0
Day                     0
Country                 0
Region                  0
ProvinceOrState       421
City                  434
Latitude             4556
Longitude            4557
Location           126196
Summary             66129
Attack_Type             0
Target_Type             0
Group_Name              0
Motive             131130
Weapon_Type             0
No_Of_Killed        10313
No_Of_Wounded       16311
Add_Notes          153402
dtype: int64

In [24]:
df['No_Of_Killed']=df['No_Of_Killed'].fillna(0)
df['No_Of_Wounded']=df['No_Of_Wounded'].fillna(0)
df['Casualty']=df['No_Of_Killed']+df['No_Of_Wounded']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documen

In [25]:
df.describe()

Year          Month            Day       Latitude  \
count  181691.000000  181691.000000  181691.000000  177135.000000   
mean     2002.638997       6.467277      15.505644      23.498343   
std        13.259430       3.388303       8.814045      18.569242   
min      1970.000000       0.000000       0.000000     -53.154613   
25%      1991.000000       4.000000       8.000000      11.510046   
50%      2009.000000       6.000000      15.000000      31.467463   
75%      2014.000000       9.000000      23.000000      34.685087   
max      2017.000000      12.000000      31.000000      74.633553   

          Longitude   No_Of_Killed  No_Of_Wounded       Casualty  
count  1.771340e+05  181691.000000  181691.000000  181691.000000  
mean  -4.586957e+02       2.266860       2.883296       5.150156  
std    2.047790e+05      11.227057      34.309747      40.555416  
min   -8.618590e+07       0.000000       0.000000       0.000000  
25%    4.545640e+00       0.000000       0.000000       0.000000  
50%    4.324651e+01       0.000000       0.000000       1.000000  
75%    6.871033e+01       2.000000       2.000000       4.000000  
max    1.793667e+02    1570.000000    8191.000000    9574.000000

## Important Observation

1. The data consists of terrorist activites ranging from the year: 1970 to 2017
2. Maximum number of people killed in  an event is: 1570
3. Maximum number of people wounded in an event is : 8191
4. Maximum number of casualty in an event were : 9574